From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Jonas Karlman <jonas@kwiboo.se>
Date: Mon, 3 Aug 2026 19:09:42 +0000
Subject: clk: rockchip: rk3506: Fix CLK_SARADC set rate issues

The set_rate ops for the CLK_SARADC clock in RK3506 clock driver
force use of 32 KHz or 400 KHz rates when any requested rate is
divisible with 32 KHz or 400 KHz.

Remove the use of a forced divisor for the 32 KHz or 400 KHz rates to
allow use of a rate divisible with 32 KHz or 400 KHz.

Signed-off-by: Austin Lane <vidplace7@gmail.com>
---
 drivers/clk/rockchip/clk_rk3506.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/clk/rockchip/clk_rk3506.c b/drivers/clk/rockchip/clk_rk3506.c
index 111111111111..222222222222 100644
--- a/drivers/clk/rockchip/clk_rk3506.c
+++ b/drivers/clk/rockchip/clk_rk3506.c
@@ -465,10 +465,10 @@ static ulong rk3506_saradc_set_rate(struct rk3506_clk_priv *priv, ulong clk_id,
 
 	if (32000 % rate == 0) {
 		sel = CLK_SARADC_SEL_32K;
-		div = 1;
+		div = DIV_ROUND_UP(32000, rate);
 	} else if (400000 % rate == 0) {
 		sel = CLK_SARADC_SEL_400K;
-		div = 1;
+		div = DIV_ROUND_UP(400000, rate);
 	} else {
 		sel = CLK_SARADC_SEL_24M;
 		div = DIV_ROUND_UP(OSC_HZ, rate);
-- 
Armbian

